.pnpmfile.cjs

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+function readPackage(pkg, context) {
+  if (pkg.dependencies["@types/react"] === "*") {
+    pkg.dependencies["@types/react"] = undefined;
+    pkg.peerDependencies["@types/react"] = "*";
+  }
+  if (pkg.peerDependencies["react"] !== undefined && pkg.peerDependencies["@types/react"] === undefined) {
+    pkg.peerDependencies["@types/react"] = "*";
+  }
+  if (pkg.name === "@itwin/map-layers-formats") {
```

Comment:
It is fixed in core and 4.3.0 have correct peer deps but in this repo we are installing 4.1.2

Reply:
couldnt we just update to pull in 4.3.0 then? guess we cant use `pnpm update` like we typically would with the resolution mode set to lowest direct anymore?

Reply:
Bumped dev deps to 4.3.0. I also switched pnpm to not use shared lock file for whole repo. Now each package has it's own `node_modules` and lock file. This makes `pnpm install` run a little longer but each package can specify different dependency versions and we don't need `pnpmfile` to work around problems like different react type versions and etc.

Reply:
does setting `share-workspace-lockfile` to `false` defeat the purpose of the monorepo?

Reply:
im kinda split on this decision, i see how this is beneficial in the repo, but feel like we lose some of the monorepo benefits. resolving issue for now, but i want to think about this further and we might need to go back to shared lockfile

Reply:
I wouldn't say that `share-workspace-lockfile` set to `false` defeats the purpose of the monorepo. If only reason why these packages were put to this monorepo was to use shared dependencies between them, then yes it defeats the purpose. However, I don't think that this monorepo should use shared dependencies between packages because it does not enforce to use same versions of those dependencies. 

I think it still worth having these packages in a monorepo because we have a `test-viewer` for debugging and testing widgets locally. All packages use same CI and release pipelines so there is no need to maintain separate pipelines in case packages are split to different repos. Only difference is that each of them now have it's on dependencies installed which should also prevent build failures encountered when dependencies are bumped in one of the packages.